src/lib/DefaultRuleManager.ts

```diff
@@ -18,6 +19,11 @@ import {uxEvents} from './ScannerEvents';
 Messages.importMessagesDirectory(__dirname);
 const messages = Messages.loadMessages('@salesforce/sfdx-scanner', 'DefaultRuleManager');
 
+type ExtendedRunDescriptor = {
+	engine: RuleEngine;
+	descriptor: EngineExecutionDescriptor;
```

Comment:
The point of the `EngineExecutionDescriptor` type is that it's all the values fed into an engine's `#run` method, and adding the engine to that object type felt like it would cause confusion. So I added this alias type instead.